src/libs/NetworkConnection.js

```diff
@@ -69,40 +43,29 @@ function listenForReconnect() {
         setOfflineStatus(!state.isConnected);
     });
 
-    // When the app is in the background Pusher can still receive realtime updates
-    // for a few minutes, but eventually disconnects causing a delay when the app
-    // returns from the background. So, if we are returning from the background
-    // and we are online we should trigger our reconnection callbacks.
-    AppState.addEventListener('change', setAppState);
-    isListeningToAppStateChanges = true;
-
     // When a device is put to sleep, NetInfo is not always able to detect
     // when connectivity has been lost. As a failsafe we will capture the time
     // every two seconds and if the last time recorded is greater than 20 seconds
     // we know that the computer has been asleep.
-    lastTime = (new Date()).getTime();
-    sleepTimer = setInterval(() => {
-        const currentTime = (new Date()).getTime();
-        if (currentTime > (lastTime + 20000)) {
-            triggerReconnectionCallbacks();
-        }
-        lastTime = currentTime;
-    }, 2000);
```

Comment:
Because Pusher event should cover it. So if the laptop is in a sleep state and it comes back to active. Pusher event will fire and so we don't need it. I will do more testing over it. Let me know other thoughts on this.

Reply:
Thanks, a lot of testing has been done. Pusher should not be the only solution we depend on. The multiple strategies are for polling redundancy. But `AppState` was too aggressive and called the reconnection callbacks too many times.

Reply:
Ok. As a fail-safe. Yeah, AppState was firing unnecessarily. Will rollback.